### PR TITLE
Support migrate & restore

### PR DESCRIPTION
**Pull request by danielfoehrKn**
**What this PR does / why we need it**:

Supports migrate and restore operations as part of the [Gardener Control Plane Migration](https://github.com/gardener/gardener/issues/1631).

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

Had to update the Gardener version to include the controller interface with restore and migrate.

**Release note**:

```improvement operator

```

### CODE REVIEW

File: pkg/controller/actuator_delete.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ import (
 	resourcemanager "github.com/gardener/gardener-resource-manager/pkg/manager"
 	extensionscontroller "github.com/gardener/gardener/extensions/pkg/controller"
 	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	gardenerutils "github.com/gardener/gardener/pkg/utils/managedresources"
```
Comment (rfranzke) [inline pkg/controller/actuator_delete.go:27]:
```suggestion
"github.com/gardener/gardener/pkg/utils/managedresources"
```

**Comment by danielfoehrKn** [comment]
I do not know why CI keeps failing with 
```
> Generate Check
go generate needs to be run:
 M example/controller-registration.yaml
make: *** [Makefile:82: check-generate] Error 1
```

locally executed there is no error.

**Comment by tim-ebert** [comment]
> I do not know why CI keeps failing with
> 
> ```
> > Generate Check
> go generate needs to be run:
>  M example/controller-registration.yaml
> make: *** [Makefile:82: check-generate] Error 1
> ```
> 
> locally executed there is no error.

You probably have a different version of helm installed than the one installed by `make install-requirements`.

**Comment by danielfoehrKn** [comment]
>  You probably have a different version of helm installed than the one installed by make install-requirements

This is such a pain -,-

**Comment by swilen-iwanow** [comment]
I didn't manage to test this today because I have a problem with my local setup for Shoots with gvisor as CRI. I am hitting a problem where pods are stuck in creation because of 
```
Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "7c145c32d5f3864aca6d9f43591123d1fc146de671b73920d153a1dd31fea2cf": failed to find plugin "calico" in path [/usr/lib/cni]
```

I will continue tomorrow.

**Pull request merged**